shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/auth/builder/loader/dialect/PostgreSQLPrivilegeLoader.java

```diff
@@ -0,0 +1,131 @@
+                    collectPrivilege(privilegeCache, resultSet);
+                }
+            }
+        }
+        fillTablePrivilege(privilegeCache, privileges);
+    }
+
+    private void fillTablePrivilege(final Map<ShardingSphereUser, Map<String, Map<String, List<PrivilegeType>>>> privilegeCache, final Map<ShardingSphereUser, ShardingSpherePrivilege> privileges) {
+        for (ShardingSphereUser user : privilegeCache.keySet()) {
+            for (String db : privilegeCache.get(user).keySet()) {
+                for (String tableName : privilegeCache.get(user).get(db).keySet()) {
+                    TablePrivilege tablePrivilege = new TablePrivilege(tableName, privilegeCache.get(user).get(db).get(tableName));
+                    ShardingSpherePrivilege privilege = privileges.get(user);
+                    if (!privilege.getDatabasePrivilege().getSpecificPrivileges().containsKey(db)) {
+                        privilege.getDatabasePrivilege().getSpecificPrivileges().put(db, new SchemaPrivilege(db));
+                    }
+                    privilege.getDatabasePrivilege().getSpecificPrivileges().get(db).getSpecificPrivileges().put(tableName, tablePrivilege);
+                }
+            }
+        }
+    }
+
+    private void collectPrivilege(final Map<ShardingSphereUser, Map<String, Map<String, List<PrivilegeType>>>> privilegeCache, final ResultSet resultSet) throws SQLException {
+        String db = resultSet.getString("table_catalog");
+        String tableName = resultSet.getString("table_name");
+        String privilegeType = resultSet.getString("privilege_type");
+        Boolean hasPrivilege = resultSet.getBoolean("is_grantable");
+        String grantee = resultSet.getString("grantee");
+        if (hasPrivilege) {
+            privilegeCache
+                    .computeIfAbsent(new ShardingSphereUser(grantee, "", ""), k -> new HashMap<>())
+                    .computeIfAbsent(db, k -> new HashMap<>())
+                    .computeIfAbsent(tableName, k -> new ArrayList<>())
+                    .add(getPrivilegeType(privilegeType));
+        }
+    }
+
+    private String getTablePrivilegeSQL(final Collection<ShardingSphereUser> users) {
+        String userList = users.stream().map(each -> each.getGrantee().getUsername())
+                .collect(Collectors.joining(", "));
+        return String.format(TABLE_PRIVILEGE_SQL, userList);
+    }
+
+    private PrivilegeType getPrivilegeType(final String privilege) {
+        switch (privilege) {
+            case "SELECT":
+                return PrivilegeType.SELECT;
+            case "INSERT":
+                return PrivilegeType.INSERT;
+            case "UPDATE":
+                return PrivilegeType.UPDATE;
+            case "DELETE":
+                return PrivilegeType.DELETE;
+            case "CREATE":
+                return PrivilegeType.CREATE;
+            case "REFERENCES":
+                return PrivilegeType.REFERENCES;
```

Comment:
How about `usage`, `connect` and others?  If needed, you can add more types in `PrivilegeType`.

Reply:
Can we transfer `Role Attributes` (login,superuser,createdb,createrole,replication,passwod,inherit) and put them info this privilege system?

Reply:
OK，I will add other `PrivilegeType`.

> I am curious there are only privileges on tables in Pg database? How about super privilege?
Can you give me some tips on how to get super privilege，I can't find it myself, it's not like MySQL.